### PR TITLE
🐛 fix: Normalization date-time should handle empty strings ""

### LINKED ISSUE #5480: Normalization date-time should handle null fields
## Tell us about the problem you're trying to solve
I'm using Mailchimp integration account.
I'm dealing with some columns with the correct type date-time but because they have null entries normalization failed casting to timestamp.
Screenshot from 2021-08-17 20-11-18

Here I modified the dbt generated sql files to try cast the second column with null entries.
Screenshot from 2021-08-17 20-12-37

## Describe the solution you’d like
A clear and concise description of what you want to see happen, or the change you would like to see

## Describe the alternative you’ve considered or used
A clear and concise description of any alternative solutions or features you've considered or are using today.

## Are you willing to submit a PR?

Remove this with your answer :-)

### PR DESCRIPTION
**Pull request by annalvova05**
## What
Fix normalization date-time should handle empty strings "" #5480 

## How
*Describe the solution*

## Recommended reading order
base-normalization/normalization/transform_catalog/stream_processor.py
base-normalization/dbt-project-template/macros/cross_db_utils/type_conversions.sql

## Pre-merge Checklist

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

### CODE REVIEW

File: airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/mysql/test_primary_key_streams/final/airbyte_tables/test_normalization/dedup_exchange_rate_scd.sql [hunk 1]
```diff
@@ -35,10 +35,11 @@ select
...
 ) as id,
     cast(currency as char) as currency,
-    cast(`date` as 
-    date
-) as `date`,
-    cast(timestamp_col as char) as timestamp_col,
+        case when `date` = '' then NULL
+        else cast(`date` as date)
+        end as `date`
+        ,
```
[the remark this answers is not in the document]
Comment (annalvova05) [reply airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/mysql/test_primary_key_streams/final/airbyte_tables/test_normalization/dedup_exchange_rate_scd.sql:41]: It's autogenerated SQL files after test running. Due to we have changes in `stream_processor.py` and `type_conversions.sql`, SQL models were re-build.

**Review by antixar — APPROVED** [review]
Great!

File: airbyte-integrations/bases/base-normalization/integration_tests/dbt_integration_test.py [hunk 2]
```diff
@@ -118,7 +118,7 @@ def setup_mysql_db(self):
         ]
         print("Executing: ", " ".join(commands))
         subprocess.call(commands)
-        time.sleep(5)
+        time.sleep(10)
```
Comment (ChristopheDuong) [inline airbyte-integrations/bases/base-normalization/integration_tests/dbt_integration_test.py:121]:
```suggestion
        print(f"....Waiting for MySql DB to start...10 sec")
        time.sleep(10)
```

**Review by ChristopheDuong — APPROVED** [review]

File: airbyte-integrations/bases/base-normalization/Dockerfile [hunk 3]
```diff
@@ -45,5 +45,5 @@ WORKDIR /airbyte
 ENV AIRBYTE_ENTRYPOINT "/airbyte/entrypoint.sh"
 ENTRYPOINT ["/airbyte/entrypoint.sh"]
 
-LABEL io.airbyte.version=0.1.46
+LABEL io.airbyte.version=0.1.47
```
Comment (Phlair) [inline airbyte-integrations/bases/base-normalization/Dockerfile:48]:
```suggestion
LABEL io.airbyte.version=0.1.49
```

File: airbyte-workers/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java [hunk 4]
```diff
@@ -47,7 +47,7 @@
 
   private static final Logger LOGGER = LoggerFactory.getLogger(DefaultNormalizationRunner.class);
 
-  public static final String NORMALIZATION_IMAGE_NAME = "airbyte/normalization:0.1.46";
+  public static final String NORMALIZATION_IMAGE_NAME = "airbyte/normalization:0.1.47";
```
Comment (Phlair) [inline airbyte-workers/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java:50]:
```suggestion
  public static final String NORMALIZATION_IMAGE_NAME = "airbyte/normalization:0.1.49";
```

File: docs/understanding-airbyte/basic-normalization.md [hunk 5]
```diff
@@ -294,6 +294,7 @@ Therefore, in order to "upgrade" to the desired normalization version, you need
 
 | Airbyte Version | Normalization Version | Date | Pull Request | Subject |
 | :--- | :---  | :--- | :--- | :--- |
+| 0.29.17-alpha | 0.1.47 | 2021-09-24 | [#6379](https://github.com/airbytehq/airbyte/pull/6379) | Handle empty string for date and date-time format |
```
Comment (Phlair) [inline docs/understanding-airbyte/basic-normalization.md:297]:
```suggestion
| 0.29.17-alpha | 0.1.49 | 2021-09-24 | [#6379](https://github.com/airbytehq/airbyte/pull/6379) | Handle empty string for date and date-time format |
```

**Review by Phlair — COMMENTED** [review]
Hey @annalvova05, just been looking at normalization stuff and saw this PR, next version for publish needs to be 0.1.50 😄

**Comment by Phlair** [comment]
> > Hey @annalvova05, just been looking at normalization stuff and saw this PR, next version for publish needs to be 0.1.49 😄
> 
> Wrong! I just published it...

Just realised! I noticed [because of this](https://github.com/airbytehq/airbyte/pull/6763#issuecomment-936669050). Seems like we're starting to get some normalization version bump overload with all the ssh / ssl changes.

**Comment by annalvova05** [comment]
@Phlair thanks, I saw new versions. Just wait for this [PR](https://github.com/airbytehq/airbyte/pull/6079) and will bump the normalization version for my issue :)

**Pull request merged**